### PR TITLE
Remove open

### PR DESCRIPTION
**Pull request by Halbaroth**
This PR removes most of the global opens in the Alt-Ergo library. 
They are replaced by local opens, aliases and explicit type annotations.

### CODE REVIEW

File: src/lib/frontend/typechecker.ml [hunk 1]
```diff
@@ -126,12 +122,12 @@ module Types = struct
             let vars = check_number_args loc lty ty in
             Ty.instantiate vars lty ty
           with Not_found ->
-            typing_error (UnknownType s) loc
+            Errors.typing_error (Errors.UnknownType s) loc
```
Comment (Stevendeo) [inline src/lib/frontend/typechecker.ml]: Either write `Errors.typing_error (UnknownType s)` if possible, or  `Errors.(typing_error (UnknownType s))`

File: src/lib/frontend/typechecker.ml [hunk 2]
```diff
@@ -126,12 +122,12 @@ module Types = struct
...
       end
 
   let add_decl ~recursive env vars id body loc =
     if MString.mem id env.to_ty && not recursive then
-      typing_error (ClashType id) loc;
+      Errors.typing_error (Errors.ClashType id) loc;
```
Comment (Stevendeo) [inline src/lib/frontend/typechecker.ml]: Same

File: src/lib/frontend/typechecker.ml [hunk 3]
```diff
@@ -559,14 +564,17 @@ let rec type_term ?(call_from_type_form=false) env f =
         let j = int_of_string j in
         match tye with
         | Ty.Tbitv n ->
-          if i>j then typing_error (BitvExtract(i,j)) loc;
-          if j>=n then typing_error (BitvExtractRange(n,j) ) loc;
+          if i>j then
+            Errors.typing_error (BitvExtract(i,j)) loc;
```
Comment (Stevendeo) [inline src/lib/frontend/typechecker.ml]: There is no need for a new line, even with`Errors.` it is less than 80 chars

Comment (Stevendeo) [reply src/lib/frontend/typechecker.ml]: There are other places where it is not necessary, I'm just mentioning it here.

File: src/lib/reasoners/ac.ml [hunk 4]
```diff
@@ -198,7 +195,7 @@ module Make (X : Sig.X) = struct
     Timers.exec_timer_pause Timers.M_AC Timers.F_make;
     x
 
-  let is_mine_symb sy _ = get_no_ac() == false && Sy.is_ac sy
+  let is_mine_symb sy _ = Options.get_no_ac() == false && Sy.is_ac sy
```
Comment (Stevendeo) [inline src/lib/reasoners/ac.ml]: `... == false` -> `not ...`
I know this is not the point of the PR, but still...

File: src/lib/reasoners/ccx.ml [hunk 5]
```diff
@@ -281,7 +279,7 @@ module Main : S = struct
           let ex = List.fold_left2 (explain_equality env) Ex.empty xs1 xs2 in
           let a = E.mk_eq ~iff:false t1 t2 in
           Debug.congruent a ex;
-          Q.push (LTerm a, ex, Th_util.Other) facts.equas
+          Q.push (Sig_rel.LTerm a, ex, Th_util.Other) facts.Sig_rel.equas
```
Comment (Stevendeo) [inline src/lib/reasoners/ccx.ml]: In the next functions, you made the choice to annotate `facts` with its precise type; why not in this function as well?

File: src/lib/reasoners/ccx.ml [hunk 6]
```diff
@@ -579,7 +577,7 @@ module Main : S = struct
     else  {env with uf = Uf.distinct env.uf lr ex}
 
   let rec assume_equalities env choices facts =
-    if Q.is_empty facts.equas then env, choices
+    if Q.is_empty facts.Sig_rel.equas then env, choices
```
Comment (Stevendeo) [inline src/lib/reasoners/ccx.ml]: Same here

File: src/lib/reasoners/enum_rel.ml [hunk 7]
```diff
@@ -26,17 +26,14 @@
...
 module A  = Xliteral
 module L  = List
 module Hs = Hstring
 
 type 'a abstract = 'a Enum.abstract = Cons of Hs.t * Ty.t |  Alien of 'a
 
 module X = Shostak.Combine
-module Sh = Shostak.Enum
-open Sh
+module Th = Shostak.Enum
```
Comment (Stevendeo) [inline src/lib/reasoners/enum_rel.ml]: `Th` could be confused with `Theory`, why changing its name?

File: src/lib/reasoners/fun_sat.ml [hunk 8]
```diff
@@ -179,10 +175,12 @@ module Make (Th : Theory.S) : Sat_solver_sig.S = struct
     unit_facts_cache : (E.gformula * Ex.t) ME.t ref;
   }
 
+  let all_models_sat_env = ref None
```
Comment (Stevendeo) [inline src/lib/reasoners/fun_sat.ml]: Rebase artifact?

File: src/lib/reasoners/matching.ml [hunk 9]
```diff
@@ -26,15 +26,10 @@
 (*                                                                            *)
 (******************************************************************************)
 
-open Format
-open Options
-open Matching_types
```
Comment (Stevendeo) [inline src/lib/reasoners/matching.ml:31]: Given the size of the module name, an alias could be given to lighten the code : 
`module MT = Matching_types`

File: src/lib/structures/errors.ml [hunk 10]
```diff
@@ -26,8 +26,6 @@
 (*                                                                            *)
 (******************************************************************************)
 
-open Format
```
Comment (Stevendeo) [inline src/lib/structures/errors.ml]: I'm not sure removing this open is relevant as you open it in almost every other function.

File: src/lib/structures/profiling.ml [hunk 11]
```diff
@@ -536,7 +533,8 @@ let print_instances_generation forced _steps fmt _timers =
         ) insts
     in
     List.iter
-      (fun (name, i, card, r) ->
+      (let fprintf = Format.fprintf in
```
Comment (Stevendeo) [inline src/lib/structures/profiling.ml]: Why not `let open Format in` ?

File: src/lib/structures/profiling.ml [hunk 12]
```diff
@@ -623,78 +621,88 @@ let switch fmt =
...
-  for _ = 0 to 206 do fprintf fmt "-" done;
-  fprintf fmt "|@.";
-  fprintf fmt "%s " (string_resize "" 13);
-  List.iter
-    (fun m ->
-       fprintf fmt "| %a  " float_print (Timers.get_sum timers m))
-    Timers.all_modules;
-  fprintf fmt "| GTimer %a |@." float_print (Options.Time.value())
+let float_print =
+  let fprintf = Format.fprintf in
```
Comment (Stevendeo) [inline src/lib/structures/profiling.ml]: Same here

File: src/lib/structures/profiling.ml [hunk 13]
```diff
@@ -623,78 +621,88 @@ let switch fmt =
...
-    print_stats true steps fmt timers;
-    fprintf fmt "@.";
-    mode := Timers;
-    print_timers true steps fmt timers;
-    fprintf fmt "@.";
-    timers_table true fmt timers;
-    fprintf fmt "@.";
-    print_instances_generation true steps fmt timers;
+let line_of_module =
+  let fprintf = Format.fprintf in
```
Comment (Stevendeo) [inline src/lib/structures/profiling.ml]: I'm stopping here, but there are others

File: src/lib/structures/ty.ml [hunk 14]
```diff
@@ -81,100 +78,102 @@ let assoc_destrs hs cases =
...
-          let first = ref true in
-          List.iter
-            (fun (s, t) ->
-               fprintf fmt "%s%s : %a" (if !first then "" else "; ")
-                 (Hstring.view s) (print body_of) t;
-               first := false
-            ) lbls;
-          fprintf fmt "}"
+  let rec print =
+    let fprintf = Format.fprintf in
```
Comment (Stevendeo) [inline src/lib/structures/ty.ml]: Here again; if you choose aliasing fprintf it is alright, but you should do it everywhere (sometimes you opened Format)

File: src/lib/util/steps.ml [hunk 15]
```diff
@@ -102,10 +99,10 @@ let incr k =
       (* Since n refers to the number of terms sent to the theories no
        * multiplier is needed here *)
       if n < 0 then
-        run_error (Invalid_steps_count n);
+        Errors.run_error (Errors.Invalid_steps_count n);
```
Comment (Stevendeo) [inline src/lib/util/steps.ml]: `Errors.Invalid...` -> `Invalid...`

File: src/lib/util/steps.ml [hunk 16]
```diff
@@ -115,7 +112,7 @@ let incr k =
         else if !steps > 0 then !steps
         else steps_bound
       in
-      run_error (Steps_limit n)
+      Errors.run_error (Errors.Steps_limit n)
```
Comment (Stevendeo) [inline src/lib/util/steps.ml]: Same

**Review by Stevendeo — CHANGES_REQUESTED** [review]

File: src/lib/reasoners/matching.ml (hunk 9 above)
Comment (Halbaroth) [reply src/lib/reasoners/matching.ml:31]: It is nice idea but I plan to improve the alias convention in another PR. For this one, I will rename the module itself.

File: src/lib/structures/profiling.ml (hunk 11 above)
Comment (Halbaroth) [reply src/lib/structures/profiling.ml]: it's a matter of taste. If you need only one function in a module, why import it completely? It's pity that Ocaml cannot open partially a module...

File: src/lib/reasoners/fun_sat.ml (hunk 8 above)
Comment (Halbaroth) [reply src/lib/reasoners/fun_sat.ml]: Yep!  What is wrong?

**Review by Stevendeo — APPROVED** [review]

**Pull request merged**